llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -381,12 +464,39 @@ bool InterBlockScheduling::updateFixPoint(BlockState &BS) {
     // Iterate on CurMBB
     return false;
   }
+  BS.setScheduled();
+
+  int &II = BS.FixPoint.II;
+  if (II) {
+    if (BS.isPipelined()) {
+      return true;
+    }
 
+    // We cut off at the load latency because the main benefit from
+    // post-pipelining comes from wrapping latencies over multiple stages.
+    // This will only work if the stages are small enough. If we miss
+    // opportunities because of this, we should try to get the Prepipeliner
+    // to accept it.
+    if (II >= 7) {
+      // Fall back to the loop schedule. Note that we can only have II != 0
+      // after the loop schedule has stabilized.
+      BS.setPipeliningFailed();
+      return true;
```

Comment:
Doesn't `true` mean that we have converged and do not need to iterate on `MBB` again? In that case don't we want to return `false` to try the "normal" loop scheduling?

Reply:
Hmm looking around, it seems the meaning of the return value is quite overloaded and tied to `isPipelined()` / `isScheduled()`. Maybe `updateFixPoint` could return a `enum State { Failed, TryLoopScheduling, TryModuloScheduling, Done }`?

Reply:
Yeah, makes sense. And that enumeration should replace the different flags in blockstate. I'm still hinging on running the scheduling algorithm as one big fixed-point iteration of all blocks, and then that state determines what to do on enterBlock etc.